### DISCUSSION TITLE
Terragrunt directory structure re-org to support categories

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#545] [Help] [answered]
If I have an infrastructure-live repository setup like this:
```
├── dev
│   ├── aurora
│   ├── eks
│   ├── istio
│   └── vpc
└── prod
    ├── aurora
    ├── eks
    ├── istio
    └── vpc
```

And would like to go to something like this for more straightforward navigation with category subdirectories:
```
├── dev
│   ├── databases
│   │   └── aurora
│   ├── networking
│   │   └── vpc
│   └── services
│       ├── eks
│       └── istio
└── prod
    ├── databases
    │   └── aurora
    ├── networking
    │   └── vpc
    └── services
        ├── eks
        └── istio
```

Is there any guidance to do this to keep the state intact? 

---

<ins datetime="2022-08-24T21:44:03Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109166">Tracked in ticket #109166</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
See https://github.com/gruntwork-io/knowledge-base/discussions/510 for an approach to handle folder refactors with Terragrunt.
